src/mlpack/methods/hmm/hmm_impl.hpp

```diff
@@ -143,6 +144,17 @@ double HMM<Distribution>::Train(const std::vector<arma::mat>& dataSeq)
       for (size_t j = 0; j < transition.n_cols; ++j)
         newLogInitial[j] = math::LogAdd(newLogInitial[j], stateLogProb(j, 0));
 
+      // Define a variable to store the value of log-probability for data.
+      arma::mat logProbs(dataSeq[seq].n_cols, transition.n_rows);
+      // Save the values of log-probability to logProbs.
+      for (size_t i = 0; i < transition.n_rows; i++)
+      {
+        // Define arma::vec to store log-probability for any dataSeq column.
+        arma::vec logProb;
+        emission[i].LogProbability(dataSeq[seq], logProb);
+        logProbs.unsafe_col(i) = logProb;
```

Comment:
Actually you can simplify this even more, with `emission[i].LogProbability(dataSeq[seq], logProbs.unsafe_col(i))`. :)